File: CS203 Data Structures and Algorithm Analysis/Template/Misc/Hanoi.cpp
#include <iostream>
using namespace std;

void hanoi(int n, char A,char B,char C){
  if(n==1){
    cout<<A<<" -> "<<C<<endl;//move A to C
  } else{
    hanoi(n-1,A,C,B);// move A to B, using C
    cout<<A<<" -> "<<C<<endl;//then move A to C
    hanoi(n-1,B,A,C);//move B to C, using A
  }
}

int main(){
    int n;
    cin>>n;
    hanoi(n,'A','B','C');
    return 0;
}

File: CS203 Data Structures and Algorithm Analysis/Template/Misc/Hanoi.py
i = 1

def move(n, fr, to):
    global i
    print("第%d步：将%d号盘子从%s -> %s"%(i, n, fr, to))
    i += 1

def hanoi(n, A, B, C):
    if n == 1:
        move(1, A, C)
    else:
        hanoi(n - 1, A, C, B)
        move(n, A, C)
        hanoi(n - 1, B, A, C)
if __name__ == "__main__":
    try:
        n=int(input('please input a integer :'))
        print('移动步骤：')
        hanoi(n,'A','B','C')
    except ValueError :
        print('please input a integer n(n>0)!')
